### PR TITLE
Add integer check for deserializer

### PR DESCRIPTION
**Pull request by vessd**
Fix a problem with untagged enum deserialization

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -324,6 +324,37 @@ impl<'a> Bytes<'a> {
...
     }
 
+    pub fn next_bytes_is_float(&self) -> bool {
+        if let Some(byte) = self.peek() {
+            let mut skip = 0;
+            match byte {
+                b'+' | b'-' => skip = 1,
+                b'0' => {
+                    if self.bytes.get(1)
+                        .filter(|&&b| "box".contains(b as char))
```
Comment (kvark) [inline src/parse.rs]: why are we parsing "box"?

Comment (vessd) [reply src/parse.rs]: if a leading 0 is present, then we check if it is binary, octal or hexadecimal
`0b`, `0o` or `0x`

the parser provides such case, but I'm not sure that it is reachable right now
https://github.com/ron-rs/ron/blob/master/src/parse.rs#L467

Comment (vessd) [reply src/parse.rs]: I realized that this is not necessary

**Pull request merged**